### DISCUSSION TITLE
Portal alternative to `com.canonical.Unity.LauncherEntry`

### DISCUSSION THREAD

**Discussion by JakobDev** [flatpak/xdg-desktop-portal#1276] [New Portals]
[com.canonical.Unity.LauncherEntry](https://wiki.ubuntu.com/Unity/LauncherAPI#Low_level_DBus_API:_com.canonical.Unity.LauncherEntry) is a DBus Interface original created by Canonical for Unity, but is today supported on other Desktop such as KDE too. It allows Programs to show additional Information in the Window Icon. These are Progress, a Number and if the Program needs your attention.

This is a really useful Feature. If you e.g. download a huge game using Heroic Games Launcher (it actually supports this Interface) you can see the Progress directly in the Window Icon without needing to switch to the Window. Message Apps can show the Count of Unread Messages.

It is currently supported out of the Box in Qt and Electron (it needs libunity installed to work) and probably others too.

The Problem is, that this API is not well known, so many Programmers don't know it even exists. The official documentation also states that the DBus interface could break at any time (which never happened) and that you should use libunity (which is deprecated).

These Features are also supported in other Operating Systems such as Windows and macOS too.

I think we should create a Portal for this functionality. This would make it more official. It would be supported by more Programs and Desktops that the current Interface. It would also allow to Sandbox this. You can use the current Interface with any Window Icon you like. The Portal would just allow the own Window icon of the Program.

The new Portal would be optional. Programs can tell the Desktop to show Information in the Icon, but it's up to the Desktop if he wants to support it. The Desktop is free to ignore that Request completely.

**Comment by TingPing** [comment]
As with any portal that is involved with a UI, this needs to be a pattern widely supported.

As I understand it this means KDE and Ubuntu has an extension for GNOME, are there any others?

**Comment by Mikenux** [comment]
About progress, we saw this in https://github.com/flatpak/xdg-desktop-portal/discussions/1236.

The thing would be to have it in the notification portal. The same would apply to counting and urgency. I mentioned Julian Sparber, since they is the assignee of https://github.com/flatpak/xdg-desktop-portal/issues/983, but as you will see, there is no news.
